CHANGELOG.unreleased.md

```diff
@@ -1167,6 +1167,31 @@ new Map([
 ]);
 ```
 
+#### TypeScript: Keep semi for a class property before index signature when no-semi is enabled ([#6728] by [@sosukesuzuki])
+
+Attempting to format the stable ver output again results in a syntax error.
+
+<!-- prettier-ignore -->
+```ts
+// Input
+export class User {
+  id: number = 2;
+  [key: string]: any
+}
+
+// Output (Prettier stable)
+export class User {
+  id: number = 2
+  [key: string]: any
+}
+
+// Output (Prettier master)
+export class User {
+  id: number = 2;
+  [key: string]: any
```

Comment:
Should the semicolon be placed at the beginning of this line, like we usually do with no-semi?

Reply:
It would probably be more idiomatic, however Prettier already formats computed class fields and methods like this.

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgMoQC2cAwhtjsADpQ504DaVI6ATq+gJ7MC6OAvIx416OANxMQmGKwCWUAOa8BOZtNbMaAXxAAaEBAAOMWdEzJQbVhADuABTYJzKdADcIsgCZ6QAIw5gANZwMHiG6GDyCsgyAK5w+gAWMISoAOqJsvCY4WBweE5Zsq5ZnMjg2D7ymHCsMHYcCoToyABm6Kg1+gBWmAAeAEIBwaHoxAAy8nBtHV0gvX14UahwAIqxEPAznQkg4aw1rOW+6L5wqD6GcrBpXjCJyAAcAAz6VxA1aRyG5Vdwh65pvoAI4beANIzOFiYAC0UDgcE8iJ8rDgoNkqIa6CaLSQ7R2+hqhFk2zmmGWazB0zxs12MFOt0892QACZ9DJ0LJUFESERmuUoNAgSBYjUACqnZz4mpaLRAA)
```sh
--parser babel
--no-semi
```

**Input:**
```jsx
class SomeClass {
    ["array"] = []
    ;["string"] = "str"
}
```

**Output:**
```jsx
class SomeClass {
  ["array"] = [];
  ["string"] = "str"
}

```

Reply:
Ok.